.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,28 @@
+name: "Linting and Testing"
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, ready_for_review]
+
+jobs:
+  lint-and-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: current
+          cache: npm
+
+      - name: Install
+        run: npm ci
+
+      - name: Prettier & ESLint
+        run: npm run lint
+
+      - name: Unit tests
+        run: npm run test:unit
```

Comment:
Do we need to set `process.env.CI` manually anywhere in order to ensure it doesn't run in [watch mode](https://vitest.dev/guide/features.html#watch-mode)?

Reply:
GitHub automatically sets it as of 2020: https://github.blog/changelog/2020-04-15-github-actions-sets-the-ci-environment-variable-to-true/